File: kotlin/src/main/kotlin/be/steinsomers/bron_kerbosch/Debug.java
package be.steinsomers.bron_kerbosch;

import org.jetbrains.annotations.NotNull;

import java.util.function.BooleanSupplier;

public final class Debug {
    public static void Assert(@NotNull BooleanSupplier condition) {
        assert condition.getAsBoolean();
    }
}

File: kotlin/src/main/kotlin/be/steinsomers/bron_kerbosch/Debug.kt
package be.steinsomers.bron_kerbosch

// Source - https://stackoverflow.com/a/79854947

@PublishedApi
internal object Debug {
    @JvmField
    @PublishedApi
    internal val ENABLED: Boolean = javaClass.desiredAssertionStatus()

    inline fun assert(lazyValue: () -> Boolean) {
        if (ENABLED) {
            val value = lazyValue()
            if (!value) {
                throw AssertionError()
            }
        }
    }

    inline fun assert(lazyValue: () -> Boolean, lazyMessage: () -> Any) {
        if (ENABLED) {
            val value = lazyValue()
            if (!value) {
                val message = lazyMessage()
                throw AssertionError(message)
            }
        }
    }
}